src/main/scala/is/hail/utils/richUtils/RichIndexedRowMatrix.scala

```diff
@@ -87,12 +88,33 @@ class RichIndexedRowMatrix(indexedRowMatrix: IndexedRowMatrix) {
 
     new BlockMatrix(new EmptyPartitionIsAZeroMatrixRDD(blocks), blockSize, rows, cols)
   }
-
+  
+  def writeAsBlockMatrix(uri: String, blockSize: Int) {
+    val rows = indexedRowMatrix.numRows()
+    val cols = indexedRowMatrix.numCols()
+    
+    val hadoop = indexedRowMatrix.rows.sparkContext.hadoopConfiguration
+    hadoop.mkDir(uri)
+    
+    // write blocks
+    hadoop.mkDir(uri + "/parts")
+    val blockCount = new WriteBlocksRDD(indexedRowMatrix, uri, blockSize).reduce(_ + _)
+    val gp = GridPartitioner(blockSize, rows, cols)
```

Comment:
This is getting constructed twice.  You want to pass it in to WriteBlocksRDD or refer to WriteBlockRDD's `gp`?